src/en/bookwalker/src/eu/kanade/tachiyomi/extension/en/bookwalker/BookWalker.kt

```diff
@@ -434,21 +427,42 @@ class BookWalker : ConfigurableSource, ParsedHttpSource(), BookWalkerPreferences
                 .asJsoup()
                 .let { validateLogin(it) }
 
-            // It generally doesn't matter which chapter we take the description from, but for
-            // series that release in volumes we want the earliest one, which will _usually_ be the
-            // last one on the page. With that said, it's not worth it to paginate in order to find
-            // the earliest volume, and volume releases don't usually have 60+ volumes anyways.
-            val firstItem = seriesPage.selectFirst(".o-tile:not(:has(.a-ribbon-pre-order)) .a-tile-ttl a")
-            val uuid = firstItem!!.absUrl("href").substringAfter("/de").substringBefore("/")
-            val bookUpdate = fetchBookUpdate(uuid)
+            val validChapters = seriesPage.select(".o-tile:not(:has($TILE_PREORDER_SELECTOR)) .a-tile-ttl a")
+
+            suspend fun linkElementToBookUpdate(link: Element): BookUpdateDto {
+                return link
+                    .absUrl("href")
+                    .substringAfter("/de")
+                    .substringBefore("/")
```

Comment:
Can you use `OkHttp`'s `HttpUrl` instead? Get the index of segment `de` and use the next segment. Or get the segment at position x, if it's static.